### PR TITLE
To/#158 osm model

### PR DESCRIPTION
**Pull request by t-ober**
Proposal for #158 

Will most likely need some more extension later on.

### CODE REVIEW

**Comment by t-ober** [comment]
Did not change `QunatityUtils` or `QuantityUtilsSpec` maybe they show up in the diff because of adding spotlessApply to the scala.

File: src/main/scala/edu/ie3/util/osm/OsmModel.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  /** Extract land uses from a list of ways
+    *
+    * @param ways
+    *   the ways to extract from
+    * @return
+    *   all ways that represent land uses
+    */
+  def extractLandUses(ways: List[Way]): List[ClosedWay] = {
+    ways.collect {
+      case way: ClosedWay if way.containsKeyValuePair(landuse, landUseValues) =>
```
Comment (ckittl) [inline src/main/scala/edu/ie3/util/osm/OsmModel.scala:64]: Here again. Does the restriction to closed ways make sense?

File: src/main/scala/edu/ie3/util/osm/TagUtils.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    // Fixme: check whether to use or delete "greenfield" and "meadow"
+    val landUseValues = Set(
+      "residential",
+      "commercial",
+      "retail",
+      // "greenfield",
+      // "meadow",
+      "farmyard"
+    )
```
Comment (ckittl) [inline src/main/scala/edu/ie3/util/osm/TagUtils.scala:44]: As discussed orally: Please leave out here and implement in OSMoGrid

File: src/main/scala/edu/ie3/util/osm/OsmModel.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  /** Extract high ways from a list of ways.
+    *
+    * @param ways
+    *   the ways to extract from
+    * @return
+    *   all ways that represent highways
+    */
+  def extractHighways(ways: List[Way]): List[Way] = {
+    ways.filter(way => way.containsKeyValuePair(highway, highWayValues))
```
Comment (ckittl) [inline src/main/scala/edu/ie3/util/osm/OsmModel.scala:52]: As discussed orally: If the "allowed" values provided in `TagUtils` are not in this project anymore, the outline of the method needs adaption.

File: src/main/scala/edu/ie3/util/osm/OsmModel.scala (hunk 1 above)
Comment (ckittl) [inline src/main/scala/edu/ie3/util/osm/OsmModel.scala:64]: As discussed orally: If the "allowed" values provided in `TagUtils` are not in this project anymore, the outline of the method needs adaption.

**Review by ckittl — CHANGES_REQUESTED** [review]

File: docs/OsmModel.puml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    ways: List[Ways]
+    relations: List[Relation]
+    capturedArea: ???
+    {static} extractBuildings(): List[ClosedWay]
+    {static} extractLandUses(): List[ClosedWay]
+    {static} extractHighWays(): List[OpenWay]
+}
+note left of OsmModel::"getHighWays(): List[OpenWay]"
+    If roundabouts are closed ways this might not work
+end note
```
[the remark this answers is not in the document]
Comment (t-ober) [reply docs/OsmModel.puml:14]: If a roundabout is considered a `ClosedWay` we can't return list of `OpenWay`

File: docs/OsmModel.puml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+getCenter(): LatLon
+}
+note left: a Way is closed if the first and last nodes \nare identical
+
+Class Relation {
+    elements: List[RelationElement]
+}
+
+Class RelationElement {
+    role: String
```
[the remark this answers is not in the document]
Comment (t-ober) [reply docs/OsmModel.puml:47]: It's an optional descriptive field which is user-defined so that won't work

**Review by ckittl — APPROVED** [review]

**Pull request merged**